### DISCUSSION TITLE
Group chat?

### DISCUSSION THREAD

**Discussion by OliviaOliveiira** [innightwolfsleep/llm_telegram_bot#88] [Q&A]
Can the bot operate in group chats? I tried making it an admin, but still, no reply. Or I have to tag the bot to operate with it?

**Comment by OliviaOliveiira** [comment]
> Can the bot operate in group chats? I tried making it an admin, but still, no reply. Or I have to tag the bot to operate with it?

It does work in the group chat, but only if the user list is empty, when I add myself there the bot does not react, only in private messages. How do I set so that the bot answers only to specific users in the chat?

Added the group ID and now it works, however, it is available to all the users even though only my id and the group ID is in the list + other users can click the buttons. How do I restrict that?

**Reply by innightwolfsleep** [reply]
Currently, there is no properly mode. Bot rules matrix work only with user id's, not with chat id's. 

Tell me desired bot behaviour for chat, I will try to design new mode

**Reply by OliviaOliveiira** [reply]
Would be great if while in the group chat, the buttons were clickable only by either whitelisted [users in users.txt], or only by group admins, which is better but I believe harder to organize + make the bot respond to the described users (again, either admins or whitelisted in users.txt). Adding a group id in the users.txt worked,  but the bot responds to everyone there.

**Comment by innightwolfsleep** [comment]
#136 Added option for customizing usernames in group chats. Thist should helt to interact in group chat with multiply users.
In config.json:
Set user_name_template = USERNAME LASTNAME 
And add "\n" to stopping_strings
